CRM/CiviAwards/Service/AwardProfile.php

```diff
@@ -28,20 +28,30 @@ public function createProfile($caseTypeId) {
   /**
    * Create fields for a profile.
    *
-   * @param array $customFieldIds
+   * @param array $customFields
    *   Custom fields.
    * @param int $profileId
    *   Profile Id.
    */
-  public function createProfileFields(array $customFieldIds, $profileId) {
-    foreach ($customFieldIds as $customFieldId) {
+  public function createProfileFields(array $customFields, $profileId) {
+    foreach ($customFields as $customField) {
       civicrm_api3('UFField', 'create', [
         'uf_group_id' => $profileId,
-        'field_name' => 'custom_' . $customFieldId,
+        'field_name' => 'custom_' . $customField['id'],
         'field_type' => 'Activity',
-        'label' => 'Activity Field' . $customFieldId,
+        'label' => 'Activity Field' . $customField['id'],
+        'is_required' => boolval($customField['required']),
+        // Nesting api call because of a bug that seems to be if you have two profile fields
```

Comment:
If you can refine the comment here. See https://github.com/compucorp/uk.co.compucorp.civicrm.prospect/blob/master/CRM/Prospect/Setup/CreateProspectMenus.php#L48-L49 also. Just say that civi will assign the next available weight when creating a profile field irrespective of the value passed in the weight field.

Might not be a bug but how civi works with weight fields on create. Might be worth taking it up with the community one of these days but it seems to be a minor issue.